tests/Unit/CssInlinerTest.php

```diff
@@ -2785,7 +2785,7 @@ public function importantIsCaseInsensitive(): void
 
         $subject->inlineCss('p { margin: 1px !ImPorTant; }');
 
-        self::assertStringContainsString('<p style="margin: 1px !ImPorTant;">', $subject->renderBodyContent());
+        self::assertStringContainsString('<p style="margin: 1px;">', $subject->renderBodyContent());
```

Comment:
I think we then also should rename the test.

Reply:
And maybe we can use a data provider for the the `important` variants (lowercase, all-uppercase, mixed case).

Reply:
> And maybe we can use a data provider for the the `important` variants (lowercase, all-uppercase, mixed case).

Good idea.  I have done that, and adjusted the tests.  I also then found another bug - `! important` (with a space after the `!`) was not being removed as intended (though it was being handled correctly elsewhere).

The final fix involves changing the `str_replace` not to a `str_ireplace` but to a `preg_replace`.  Now Psalm complains because `preg_replace` _might_ return `null`.  So I have had to implement a wrapper to ensure that is not the case.

Do some of the other Psalm issues included in the `baseline` relate to this (we use `preg_replace` elsewhere)?  Maybe this wrapper will allow that specific issue to be addressed in those other cases.

Reply:
Seems I also need to disable a PHPCS rule, because I declared the extra exception that could be thrown.